internal/utils/miscellaneous.go

```diff
@@ -81,3 +82,33 @@ func ComputeScheduleInterval(cronSchedule string) (time.Duration, error) {
 	nextNextScheduledTime := schedule.Next(nextScheduledTime)
 	return nextNextScheduledTime.Sub(nextScheduledTime), nil
 }
+
+func getValueOrError(data map[string]string, key string) (string, error) {
+	value, ok := data[key]
+	if !ok {
+		return "", fmt.Errorf("key %s not found in data", key)
+	}
+	return value, nil
+}
+
+func parseBoolOrError(value string) (bool, error) {
+	result, err := strconv.ParseBool(value)
+	if err != nil {
+		return false, fmt.Errorf("failed to parse boolean value: %w", err)
+	}
+	return result, nil
+}
+
+// GetBoolValueOrDefault returns the boolean value for the given key from the data map,
+// and defaults to defaultValue if the key is not found or the value is not a valid boolean.
+func GetBoolValueOrDefault(data map[string]string, key string, defaultValue bool) bool {
+	value, err := getValueOrError(data, key)
```

Comment:
why should `getValueOrError` exist and why should it return an error if the key is not found?

Reply:
Ok you're right. We don't really need the extra function unnecessarily. I'll remove this.

Reply:
Fixed in https://github.com/gardener/etcd-druid/pull/996/commits/0f9cdd832d1be9e6cc36dcec8c17c1a97b0e12a9